client/www/pages/docs/storage.md

```diff
@@ -5,131 +5,555 @@ title: Storage
 Instant Storage makes it simple to upload and serve files for your app.
 You can use Storage to store images, videos, documents, and any other file type.
 
-{% callout %}
+## Storage quick start
 
-Storage is still in **beta**, but you can request access [here](https://docs.google.com/forms/d/e/1FAIpQLSdzInffrNrsYaamtH_BUe917EOpcOq2k8RWcGM19XepJR6ivQ/viewform?usp=sf_link)!
+Here's a full example of how to upload and display a grid of images
 
-{% /callout %}
+```shell {% showCopy=true %}
+npx create-next-app instant-storage --tailwind --yes
+cd instant-storage
+npm i @instantdb/react
+```
+
+Initialize your schema and permissions via the [cli tool](/docs/cli)
 
-## Uploading files
+```
+npx instant-cli@latest init
+```
 
-We use the `db.storage.upload(pathname: string, file: File)` function to upload a file.
+Now open `instant.perms.ts` and add the following permissions
 
-```tsx
-async function upload(files: FileList) {
-  const file = files[0];
-  // use the file's current name as the path
-  await db.storage.upload(file.name, file);
-  // or, give the file a custom name
-  await db.storage.upload('demo.png', file);
-  // or, put it in the `images` subdirectory
-  await db.storage.upload('images/demo.png', file);
-  // or, put it in a subdirectory for the current user,
-  // and restrict access to this file via Storage permissions
-  await db.storage.upload(`${currentUser.id}/demo.png`, file);
+```javascript {% showCopy=true %}
+// Note:
+// For production apps you should use more restrictive permissions
+// But for this quick start example we'll allow anyone to view, upload,
+// and delete files
+{
+  "$files": {
+    "allow": {
+      "view": "true",
+      "create": "true",
+      "delete": "true"
+    }
+  }
+}
+```
+
+And then replace the contents of `app/src/page.tsx` with the following code
+
+```javascript {% showCopy=true %}
+'use client';
+
+import { init } from '@instantdb/react';
+import schema from '../instant.schema';
+import React from 'react';
+
+// Types
+// ----------
+export type Image = {
```

Comment:
nit: if we are using `schema`, i.m.o we can just extract the type from it. 

For example: 

```
type File = InstaQLEntity<AppSchema, '$files'>